### PR TITLE
created Extending-TrialEmulation vignette for Data Stores

### PR DESCRIPTION
**Pull request by darkgoldenrod-cherry**

### CODE REVIEW

**Comment by gravesti** [comment]
@darkgoldenrod-cherry Could you review my additions to see if it reads ok?

File: vignettes/Extending-TrialEmulation.Rmd [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+Three classes are required implementing a model fitter:
+
+- **te_model_fitter**: Parent class. This class is virtual so no object can be created with this class. It exists to
+allow the definition of child classes.
+  - @save_path A path to a directory for saving models
+
+- **te_outcome_fitted**: Parent class. This class contains the results of the fitting an outcome model. A class 
+  inheriting from `te_outcome_fitted` must be defined for a new model fitter implementation.
+  - @model: A list containing the fitted model objects
+  - @summary:  A list of data frames containing a summary of the fitted model (`tidy`, `glace`) and the saved file
```
Comment (darkgoldenrod-cherry) [inline vignettes/Extending-TrialEmulation.Rmd:42]: Probably 'glance' on line 42? The same on line 47

File: vignettes/Extending-TrialEmulation.Rmd [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+  - @files: Paths to all available files
+  - @template: empty `data.frame`, used as a template when reading the data to preserve types and attributes
+  - @N: inherited from `te_datastore`
+- **te_datastore_datatable**: Expanded data is saved as a `data.table` in memory, only viable for smaller datasets.
+  - @data: `data.table` containing expanded data
+  - @N: inherited from `te_datastore`
+- **te_datastore_duckdb**: Expanded data is saved as a DuckDB file containing all trial periods. Reading, subsetting and
+sampling can be done efficiently with an SQL query (currently constructed with a translator helper function).
+  - @path: Path of the DuckDB file
+  - @table: The table name
```
Comment (darkgoldenrod-cherry) [inline vignettes/Extending-TrialEmulation.Rmd:155]: Does this work? As far as I remember this slot was always empty when I looked at it.

Comment (gravesti) [reply vignettes/Extending-TrialEmulation.Rmd:155]: I updated the code here, but should also add a test...

Comment (gravesti) [reply vignettes/Extending-TrialEmulation.Rmd:155]: For now the user can't change it in `save_to_duckdb()` but it should use the slot otherwise now.

**Review by gravesti — APPROVED** [review]

**Pull request merged**